# STAT 207 Lab Assignment 10 - Group Part [9 Points]



## <u>Case Study</u>: Predicting Airbnb Superhost Status

Suppose that you have a spare room in your house (in Seattle) and you are considering advertising this room to guests on Airbnb. Airbnb superhosts are considered to be experienced hosts who provide a shining example for other hosts, and extraordinary experiences for their guests. Once a host reaches Superhost status, a superhost badge will automatically appear on their listing and profile to help guests identify them. You would like to assess your chances of being named a superhost with your property.

The following dataset is a sample (assume random) of available Airbnb listings in Seattle, WA. These listings were collected in January 2016, and filtered to just contain listings from the five most popular Seattlbe neighborhoods (for Airbnb listings) and just contain listings that are either in a house or apartment property.

This dataset contains the following variables.

**Listing Information**
The dataset contains the following information about the Airbnb *listing*:
* <u>price</u>: price of the listing (per night)(in US dollars)
* <u>review_scores_rating</u>: the average rating of the listing [0,100] (100 is the best)
* <u>number_of_reviews</u>: the number of reviews for the listing
* <u>security_deposit</u>: the security deposit required for the listing (in US dollars)
* <u>cleaning_fee</u>: the cleaning fee required for the listing (in US dollars)
* <u>neighborhood</u>: the neighborhood of Seattle the listing is located in
* <u>property_type</u>: is the listing in a 'House' or 'Apartment'
* <u>room_type</u>: is the listing a 'Entire home/apt', 'Private room', or 'Shared room'
* <u>accommodates</u>: how many guests will the listing accommodate
* <u>bathrooms</u>:how many bathrooms does the listing have
* <u>beds</u>: how many beds does the listing have

**Host Information**
The dataset also contains the following information about the *host* of the given Airbnb listing:
* <u>host_is_superhost</u>: is the host a "superhost": t=True, f=False
* <u>host_has_profile_pic</u>: does the host have a profile pic in their bio: t=True, f=False
* <u>host_response_time</u>: how fast will the host respond to requests (on average)

## Research Goals

The primary research goal that we will pursue in this lab will be to build a predictive model that effectively predicts the `probability a Seattle Airbnb host is a **superhost**` for *new Seattle Airbnb listings/hosts* given some combination of the `remaining explanatory variables`.



## Points

<p>&nbsp;</p>
<table style="border: none;border-collapse: collapse;width:102pt;">
    <tbody>
        <tr>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid black;height:14.25pt;width:51pt;">Problem</td>
            <td style="color:black;font-size:15px;font-weight:700;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:general;vertical-align:bottom;border:.5pt solid black;border-left:none;width:51pt;">Points</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">1.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.5</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.1</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.2</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.75</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">0.25</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">2.4</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">3</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">3</td>
        </tr>
        <tr>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;height:14.25pt;border-top:none;">4</td>
            <td style="color:black;font-size:15px;font-weight:400;font-style:normal;text-decoration:none;font-family:Calibri, sans-serif;text-align:right;vertical-align:bottom;border:.5pt solid black;border-top:none;border-left:none;">1.5</td>
        </tr>
    </tbody>
</table>

In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

from sklearn.model_selection import cross_val_score
from sklearn.model_selection import KFold
from sklearn.model_selection import train_test_split

import statsmodels.formula.api as smf

from sklearn.linear_model import Ridge, Lasso, ElasticNet, LinearRegression
from sklearn.metrics import r2_score

from sklearn.preprocessing import StandardScaler

import warnings
from sklearn.exceptions import ConvergenceWarning

# Ignore scikit-learn ConvergenceWarnings
warnings.filterwarnings("ignore", category=ConvergenceWarning)
warnings.filterwarnings("ignore", category=UserWarning)

## 1. Basic Data Examination and Manipulation

### 1.1. Basics
1. Read the `seattle_airbnb_listings.csv` file into a dataframe.

2. Display your dataframe.

3. Show how many rows your dataframe has.



In [3]:
df = pd.read_csv("seattle_airbnb_listings.csv")
df.head()

price  review_scores_rating  number_of_reviews  security_deposit  \
0    300                   100                 24               500   
1    149                    96                 11               300   
2     95                    95                 79               150   
3    105                   100                 13               500   
4    140                    99                 30               250   

   cleaning_fee neighborhood property_type        room_type  accommodates  \
0            95  Wallingford         House  Entire home/apt             5   
1           105  Wallingford     Apartment  Entire home/apt             6   
2            40  Wallingford     Apartment  Entire home/apt             3   
3            50  Wallingford         House     Private room             2   
4            65  Wallingford         House  Entire home/apt             2   

   bathrooms  beds  host_response_time  host_response_rate host_is_superhost  \
0        1.5     3  within a few hours                 1.0                 t   
1        1.0     3      within an hour                 1.0                 f   
2        1.0     2      within an hour                 1.0                 f   
3        2.0     1  within a few hours                 1.0                 t   
4        1.0     1      within an hour                 1.0                 t   

  host_has_profile_pic host_identity_verified  
0                    t                      t  
1                    t                      t  
2                    t                      t  
3                    t                      t  
4                    t                      t

In [5]:
shape1 = df.shape[0]
print(f"The dataframe has {shape1} rows")

The dataframe has 325 rows


### 1.2. 0/1 Response Variable

Create a new 0/1 variable in your dataset in which `superhosts=1` and `nonsuperhosts=0`. 

In [9]:
df["host_is_superhost"].replace({"t": 1, "f": 0},inplace=True)

### 1.3. Train-Test-Split

Create a training dataset and a test dataset from your dataframe using a random state of `207` and a test dataset size that is a random sample of about 20% of the observations from the original dataset.

In [10]:
df_train, df_test = train_test_split(df, test_size=0.2, random_state=207)

## 2. Full Model and Test Log-Likelihood Function

In the interest of time, let's only consider using 5 potential explanatory variables in our logistic regression model, which will predict the `probability that a host is a superhost`:

* `review_scores_rating`
* `number_of_reviews`
* `neighborhood`
* `room_type`
* `property_type`

### 2.1. Fit the Model

Fit this "full" logistic regression model and show the summary output table. Use this summary output table to find the log-likelihood function value of this model and the training dataset.

The log-likelihood function value is -129

In [12]:
df = df[["review_scores_rating", "number_of_reviews", "neighborhood", "room_type", "property_type", "host_is_superhost"]]

In [13]:
log_mod = smf.logit(formula="host_is_superhost~review_scores_rating+number_of_reviews+neighborhood+room_type+property_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.496145
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      250
Method:                           MLE   Df Model:                            9
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                  0.1878
Time:                        09:43:15   Log-Likelihood:                -129.00
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 1.563e-09
===============================================================================================
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                     -24.7339      5.667     -4.365      0.000     -35.840     -13.628
neighborhood[T.Broadway]       -1.0746      0.435     -2.468      0.014      -1.928      -0.221
neighborhood[T.Fremont]        -0.9712      0.618     -1.572      0.116      -2.182       0.240
neighborhood[T.Minor]          -0.1215      0.549     -0.222      0.825      -1.197       0.954
neighborhood[T.Wallingford]    -0.3893      0.545     -0.714      0.475      -1.457       0.679
room_type[T.Private room]       0.2296      0.459      0.500      0.617      -0.671       1.130
room_type[T.Shared room]       -1.7915      1.104     -1.622      0.105      -3.956       0.373
property_type[T.House]         -0.2752      0.415     -0.662      0.508      -1.089       0.539
review_scores_rating            0.2515      0.058      4.350      0.000       0.138       0.365
number_of_reviews               0.0135      0.005      2.885      0.004       0.004       0.023
===============================================================================================
"""

### 2.2. Training Data Predictive Probabilities

Calculate the predictive probabilities for all observations in the **training** dataset with model. **Call the pandas series that contains each of these predictive probabilities `phat_train`**.

*Note: In addition to using the **.predict()** function to predict the **predictive probability** of a single observation with a logistic regression model, we can use **.predict(YOUR_DATAFRAME)** to predict the predictive probabilities for all observations in a given dataframe with a logistic regression model.*

In [21]:
phat_train = log_mod.predict(df_train)

### 2.3. Training Log-Likelihood Function Value

Assuming that your training dataframe was called `df_train`, the code below should calculate the the log-likelihood function value for your training dataset and this model. (If not, modify the code below to contain the actual name of your training dataframe).

This **log_loss()** should take take the following values:
* the actual 0/1 response variable values
* the predictive probabilities of a given model

Verify that this output here is the same as what you got in 2.1.

In [22]:
from sklearn.metrics import log_loss

-log_loss(df_train['host_is_superhost'], phat_train)*len(df_train['host_is_superhost'])

-128.99761447923498

This is the same output that we got in 2.1

### 2.4. Test Log-Likelihood Function Value

Use the code in 2.2 and 2.3 to now calculate the log-likelihood function value of the **test dataset** and this model.

In [24]:
phat_test = log_mod.predict(df_test)
-log_loss(df_test['host_is_superhost'], phat_test)*len(df_test['host_is_superhost'])

-28.68953144226903

## 3. Backwards Elimination Algorithm

Ideally, we'd like to train (with our training dataset) and select the logistic regression model that will yield the HIGHEST **test log-likelihood function (LLF) value** (or more specifically will be the LLF that is the least negative). Let's use a backwards elimination algorithm to try to help us find this model with the highest test LLF value.

*Note: Remember that the backwards elimination algorithm is just an algorithm "template". You can use this algorithm template to evaluate other types of models (like logistic regression) in attempt to find the model that has the highest metric value (like the highest test LLF value).*

In [29]:
log_mod = smf.logit(formula="host_is_superhost~review_scores_rating+number_of_reviews+neighborhood+room_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.496994
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      251
Method:                           MLE   Df Model:                            8
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                  0.1864
Time:                        09:56:53   Log-Likelihood:                -129.22
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 6.650e-10
===============================================================================================
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                     -24.7734      5.659     -4.378      0.000     -35.864     -13.682
neighborhood[T.Broadway]       -1.1119      0.432     -2.571      0.010      -1.959      -0.264
neighborhood[T.Fremont]        -1.1198      0.579     -1.935      0.053      -2.254       0.015
neighborhood[T.Minor]          -0.2904      0.485     -0.598      0.550      -1.242       0.661
neighborhood[T.Wallingford]    -0.5594      0.481     -1.164      0.244      -1.502       0.383
room_type[T.Private room]       0.1292      0.432      0.299      0.765      -0.718       0.976
room_type[T.Shared room]       -1.8679      1.094     -1.707      0.088      -4.012       0.277
review_scores_rating            0.2519      0.058      4.364      0.000       0.139       0.365
number_of_reviews               0.0135      0.005      2.884      0.004       0.004       0.023
===============================================================================================
"""

In [30]:
log_mod = smf.logit(formula="host_is_superhost~review_scores_rating+number_of_reviews+neighborhood+property_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.504332
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      252
Method:                           MLE   Df Model:                            7
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                  0.1744
Time:                        09:56:58   Log-Likelihood:                -131.13
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 1.244e-09
===============================================================================================
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                     -25.0679      5.547     -4.519      0.000     -35.939     -14.196
neighborhood[T.Broadway]       -1.0203      0.430     -2.375      0.018      -1.862      -0.178
neighborhood[T.Fremont]        -0.8252      0.613     -1.347      0.178      -2.026       0.376
neighborhood[T.Minor]           0.0588      0.542      0.109      0.914      -1.003       1.121
neighborhood[T.Wallingford]    -0.3990      0.540     -0.739      0.460      -1.457       0.659
property_type[T.House]         -0.3197      0.387     -0.827      0.408      -1.078       0.438
review_scores_rating            0.2536      0.057      4.485      0.000       0.143       0.364
number_of_reviews               0.0153      0.005      3.318      0.001       0.006       0.024
===============================================================================================
"""

In [31]:
log_mod = smf.logit(formula="host_is_superhost~review_scores_rating+number_of_reviews+room_type+property_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.513461
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      254
Method:                           MLE   Df Model:                            5
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                  0.1595
Time:                        09:57:03   Log-Likelihood:                -133.50
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 1.020e-09
=============================================================================================
                                coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------------
Intercept                   -27.2353      5.544     -4.912      0.000     -38.102     -16.369
room_type[T.Private room]     0.0928      0.443      0.209      0.834      -0.776       0.961
room_type[T.Shared room]     -1.7968      1.081     -1.662      0.097      -3.916       0.322
property_type[T.House]       -0.2014      0.332     -0.606      0.544      -0.852       0.450
review_scores_rating          0.2719      0.057      4.802      0.000       0.161       0.383
number_of_reviews             0.0132      0.005      2.812      0.005       0.004       0.022
=============================================================================================
"""

In [32]:
log_mod = smf.logit(formula="host_is_superhost~review_scores_rating+neighborhood+room_type+property_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.512400
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      251
Method:                           MLE   Df Model:                            8
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                  0.1612
Time:                        09:57:07   Log-Likelihood:                -133.22
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 2.400e-08
===============================================================================================
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                     -19.2431      4.757     -4.046      0.000     -28.566      -9.920
neighborhood[T.Broadway]       -1.0274      0.430     -2.388      0.017      -1.870      -0.184
neighborhood[T.Fremont]        -1.0922      0.608     -1.796      0.072      -2.284       0.100
neighborhood[T.Minor]          -0.1900      0.542     -0.351      0.726      -1.252       0.872
neighborhood[T.Wallingford]    -0.3653      0.538     -0.680      0.497      -1.419       0.688
room_type[T.Private room]       0.4007      0.445      0.901      0.368      -0.471       1.273
room_type[T.Shared room]       -2.1011      1.090     -1.928      0.054      -4.237       0.034
property_type[T.House]         -0.2641      0.407     -0.648      0.517      -1.063       0.534
review_scores_rating            0.1989      0.049      4.055      0.000       0.103       0.295
===============================================================================================
"""

In [33]:
log_mod = smf.logit(formula="host_is_superhost~number_of_reviews+neighborhood+room_type+property_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.553244
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      251
Method:                           MLE   Df Model:                            8
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                 0.09433
Time:                        09:57:14   Log-Likelihood:                -143.84
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 0.0002146
===============================================================================================
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                      -0.2318      0.329     -0.705      0.481      -0.876       0.413
neighborhood[T.Broadway]       -1.3502      0.418     -3.233      0.001      -2.169      -0.532
neighborhood[T.Fremont]        -1.2059      0.589     -2.048      0.041      -2.360      -0.052
neighborhood[T.Minor]          -0.0184      0.520     -0.035      0.972      -1.038       1.001
neighborhood[T.Wallingford]    -0.4858      0.517     -0.939      0.348      -1.499       0.528
room_type[T.Private room]       0.4762      0.425      1.121      0.262      -0.356       1.309
room_type[T.Shared room]       -1.7708      1.071     -1.654      0.098      -3.869       0.328
property_type[T.House]         -0.3680      0.391     -0.942      0.346      -1.133       0.397
number_of_reviews               0.0072      0.004      1.787      0.074      -0.001       0.015
===============================================================================================
"""

## 4. Final Model

Write out your final logistic regression model in equation form using the three different representations that we discussed in class.
1. The one that predicts the **probability** the host is a superhost on the left hand side.
2. The one that predicts the **odds** the host is a superhost on the left hand side.
3. The one that predicts the **log-odds** the host is a superhost on the left hand side.

Make sure that you use the appropriate notation that we discussed in class.

In [34]:
log_mod = smf.logit(formula="host_is_superhost~review_scores_rating+number_of_reviews+neighborhood+room_type+property_type",data=df_train).fit()
log_mod.summary()

Optimization terminated successfully.
         Current function value: 0.496145
         Iterations 7


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:      host_is_superhost   No. Observations:                  260
Model:                          Logit   Df Residuals:                      250
Method:                           MLE   Df Model:                            9
Date:                Wed, 27 Mar 2024   Pseudo R-squ.:                  0.1878
Time:                        09:59:23   Log-Likelihood:                -129.00
converged:                       True   LL-Null:                       -158.82
Covariance Type:            nonrobust   LLR p-value:                 1.563e-09
===============================================================================================
                                  coef    std err          z      P>|z|      [0.025      0.975]
-----------------------------------------------------------------------------------------------
Intercept                     -24.7339      5.667     -4.365      0.000     -35.840     -13.628
neighborhood[T.Broadway]       -1.0746      0.435     -2.468      0.014      -1.928      -0.221
neighborhood[T.Fremont]        -0.9712      0.618     -1.572      0.116      -2.182       0.240
neighborhood[T.Minor]          -0.1215      0.549     -0.222      0.825      -1.197       0.954
neighborhood[T.Wallingford]    -0.3893      0.545     -0.714      0.475      -1.457       0.679
room_type[T.Private room]       0.2296      0.459      0.500      0.617      -0.671       1.130
room_type[T.Shared room]       -1.7915      1.104     -1.622      0.105      -3.956       0.373
property_type[T.House]         -0.2752      0.415     -0.662      0.508      -1.089       0.539
review_scores_rating            0.2515      0.058      4.350      0.000       0.138       0.365
number_of_reviews               0.0135      0.005      2.885      0.004       0.004       0.023
===============================================================================================
"""

$\widehat{p} = \dfrac{
1
}{
1 + e^{\begin{aligned}
    &-24.7339 \\
    &- 1.0756*neighborhood[T.Broadway]\\
    &- 0.9712*neighborhood[T.Fremont]\\
    &- 0.1215*neighborhood[T.Minor]\\
    &- 0.3893*neighborhood[T.Wallingford]\\
    &+ 0.2296*room_type[T.Private room]\\
    &- 1.7915*room_type[T.Shared room]\\
    &- 0.2752*property_type[T.House]\\
    &+ 0.2515*review_scores_rating\\
    &+ 0.0135*number_of_reviews\\
    \end{aligned}
}
}$

$\widehat{odds} = \dfrac{
\widehat{p}
}{
1 - \widehat{p}
} = e^{\begin{aligned}
    &-24.7339 \\
    &- 1.0756*neighborhood[T.Broadway]\\
    &- 0.9712*neighborhood[T.Fremont]\\
    &- 0.1215*neighborhood[T.Minor]\\
    &- 0.3893*neighborhood[T.Wallingford]\\
    &+ 0.2296*room_type[T.Private room]\\
    &- 1.7915*room_type[T.Shared room]\\
    &- 0.2752*property_type[T.House]\\
    &+ 0.2515*review_scores_rating\\
    &+ 0.0135*number_of_reviews\\
    \end{aligned}
}
$

$log(\widehat{odds}) = log(\dfrac{
\widehat{p}
}{
1 - \widehat{p}
}) = \begin{aligned}
    &-24.7339 \\
    &- 1.0756*neighborhood[T.Broadway]\\
    &- 0.9712*neighborhood[T.Fremont]\\
    &- 0.1215*neighborhood[T.Minor]\\
    &- 0.3893*neighborhood[T.Wallingford]\\
    &+ 0.2296*room_type[T.Private room]\\
    &- 1.7915*room_type[T.Shared room]\\
    &- 0.2752*property_type[T.House]\\
    &+ 0.2515*review_scores_rating\\
    &+ 0.0135*number_of_reviews\\
    \end{aligned}$

*<u>Teammates:</u>*

*1. Jessica Micklin*

*2. Kaylee Cheu*

*3. Ryan Sponzilli*